### DISCUSSION TITLE
graphql/codegen/docs/getting-started/installation

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9510] [Docs Discussions]
# graphql/codegen/docs/getting-started/installation

GraphQL Code Generator is a plugin-based tool that helps you get the best out of your GraphQL stack. From back-end to front-end, GraphQL Code Generator automates the generation of typed Queries, Mutations and, Subscriptions for React, Vue, Angular, Next.js, Svelte, whether you are using Apollo Client, URQL or, React Query.

https://the-guild.dev/graphql/codegen/docs/getting-started/installation

**Comment by ndnam0109** [comment]


**Reply by ndnam0109** [reply]
/

**Reply by ndnam0109** [reply]
.

**Reply by ndnam0109** [reply]
/

**Comment by yuercl** [comment]
Error: connect ETIMEDOUT 104.16.122.175:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -60,
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '104.16.122.175',
  port: 443
}

**Reply by yuercl** [reply]
add this to local `codegen.ts`, and run commad again
```
  debug: true,
  verbose: true,
```

**Comment by Jcogollohernandez12** [comment]
Could not generate `fromJson` code for `id`.
To support the type `InvalidType` you can:
* Use `JsonConverter`
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonConverter-class.html
* Use `JsonKey` fields `fromJson` and `toJson`
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/fromJson.html
  https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/toJson.html
package:concert_plaza/src/core/domain/entities/app_models.freezed.dart:957:23
    ╷
957 │   IDFilterComparison? id;

**Comment by danmandel** [comment]
I'm running `npx graphql-code-generator init` and it freezes on the third step.

```
? What type of application are you building? Backend - API or server
? Where is your schema?: (path or url) http://localhost:4000/graphql
? Pick plugins: (Press <space> to select, <a> to toggle all, <i> to invert selection, and <enter> to proceed)
```

Doesn't react to any keypresses after that

**Reply by horiuchinobuyuki** [reply]
I got this problem. `pnpm add -D typescript` fixed it.
